dulwich/index.py

```
@@ -533,6 +554,8 @@ def changes_from_tree(
 
         def lookup_entry(path):
             entry = self[path]
+            if isinstance(entry, ConflictedIndexEntry):
```

I think this should raise an exception rather than silently defaulting to .this (which may not actually be set)

---

please also add a test for this behaviour